{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "10b477ab-7628-426e-b36b-82f06a824ae0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pyrebase as py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d34df5b1-2975-4b4c-97c8-997590855554",
   "metadata": {},
   "outputs": [],
   "source": [
    "token = {\n",
    "  \"apiKey\": \"AIzaSyC8LfrnWarDkmPTLEUCG06DYzUML0S7Bkw\",\n",
    "  \"authDomain\": \"ue22166883.firebaseapp.com\",\n",
    "  \"databaseURL\": \"https://ue22166883-default-rtdb.firebaseio.com\",\n",
    "  \"projectId\": \"ue22166883\",\n",
    "  \"storageBucket\": \"ue22166883.appspot.com\",\n",
    "  \"messagingSenderId\": \"536390400329\",\n",
    "  \"appId\": \"1:536390400329:web:14e682403a7dcea9314298\",\n",
    "  \"measurementId\": \"G-FGK3R4Z6MZ\"\n",
    "};"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "01fe0351-2701-4205-9a97-1e1c1d8c2d3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "firebase = py.initialize_app(token)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1d464284-04d6-4865-b3e0-46f9d00bceff",
   "metadata": {},
   "outputs": [],
   "source": [
    "ddbb = firebase.database()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cb8d4469-818b-49fc-b5f6-f1778e1a78ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "user= {\"nom\":\"King\",\n",
    "       \"surn\":\"Boss\",\n",
    "       \"email\":\"lopezcarlosuem@gmail.com\",\n",
    "      \"exp\":\"22166883\",\n",
    "      \"faculty\":\"AID\",\n",
    "      \"birth\":\"05-05-1990\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fafada07-5b68-403f-8337-02e9fd7ec960",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'nom': 'King',\n",
       " 'surn': 'Boss',\n",
       " 'email': 'lopezcarlosuem@gmail.com',\n",
       " 'exp': '22166883',\n",
       " 'faculty': 'AID',\n",
       " 'birth': '05-05-1990'}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "user\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "52b7becb-02cb-4eca-a569-a33fd0155ec5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'name': '-NGwSnOq1mh6vHw3eJ9I'}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ddbb.child(\"users\").push(user)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "1841cd61-06ec-4884-8427-9c59cd249c05",
   "metadata": {},
   "outputs": [],
   "source": [
    "storage=firebase.storage()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "734ecc62-74e2-4512-9073-c6c586adfade",
   "metadata": {},
   "outputs": [
    {
     "ename": "HTTPError",
     "evalue": "[Errno 404 Client Error: Not Found for url: https://firebasestorage.googleapis.com/v0/b/ue22166883.appspot.com/o?name=users/fotos.jpg] {\n  \"error\": {\n    \"code\": 404,\n    \"message\": \"Not Found.\"\n  }\n}",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pyrebase\\pyrebase.py:512\u001b[0m, in \u001b[0;36mraise_detailed_error\u001b[1;34m(request_object)\u001b[0m\n\u001b[0;32m    511\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 512\u001b[0m     \u001b[43mrequest_object\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mraise_for_status\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    513\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m HTTPError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    514\u001b[0m     \u001b[38;5;66;03m# raise detailed error message\u001b[39;00m\n\u001b[0;32m    515\u001b[0m     \u001b[38;5;66;03m# TODO: Check if we get a { \"error\" : \"Permission denied.\" } and handle automatically\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\models.py:1021\u001b[0m, in \u001b[0;36mResponse.raise_for_status\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m   1020\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m http_error_msg:\n\u001b[1;32m-> 1021\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m HTTPError(http_error_msg, response\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m)\n",
      "\u001b[1;31mHTTPError\u001b[0m: 404 Client Error: Not Found for url: https://firebasestorage.googleapis.com/v0/b/ue22166883.appspot.com/o?name=users/fotos.jpg",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn [17], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[43mstorage\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchild\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43musers/fotos.jpg\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mput\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mlangui.jpg\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pyrebase\\pyrebase.py:459\u001b[0m, in \u001b[0;36mStorage.put\u001b[1;34m(self, file, token)\u001b[0m\n\u001b[0;32m    457\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    458\u001b[0m     request_object \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrequests\u001b[38;5;241m.\u001b[39mpost(request_ref, data\u001b[38;5;241m=\u001b[39mfile_object)\n\u001b[1;32m--> 459\u001b[0m     \u001b[43mraise_detailed_error\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest_object\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    460\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m request_object\u001b[38;5;241m.\u001b[39mjson()\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pyrebase\\pyrebase.py:516\u001b[0m, in \u001b[0;36mraise_detailed_error\u001b[1;34m(request_object)\u001b[0m\n\u001b[0;32m    512\u001b[0m     request_object\u001b[38;5;241m.\u001b[39mraise_for_status()\n\u001b[0;32m    513\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m HTTPError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    514\u001b[0m     \u001b[38;5;66;03m# raise detailed error message\u001b[39;00m\n\u001b[0;32m    515\u001b[0m     \u001b[38;5;66;03m# TODO: Check if we get a { \"error\" : \"Permission denied.\" } and handle automatically\u001b[39;00m\n\u001b[1;32m--> 516\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m HTTPError(e, request_object\u001b[38;5;241m.\u001b[39mtext)\n",
      "\u001b[1;31mHTTPError\u001b[0m: [Errno 404 Client Error: Not Found for url: https://firebasestorage.googleapis.com/v0/b/ue22166883.appspot.com/o?name=users/fotos.jpg] {\n  \"error\": {\n    \"code\": 404,\n    \"message\": \"Not Found.\"\n  }\n}"
     ]
    }
   ],
   "source": [
    "storage.child(\"users/fotos.jpg\").put(\"langui.jpg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3145c45d-55ef-4fc1-97f8-c2b1e1315f94",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
